server/api/messaging.py

```diff
@@ -0,0 +1,207 @@
+    if receiver_id == None:
+        response_dict['status'] = 400
+        response_dict['message'] = "Cannot create a new message with no receiving user"
+        return jsonify(response_dict), 400
+    # Get the receiving user by getting the corresponding
+    # instance using the receiver_id passed in the request
+    message_info = {}
+    try:
+        # Sending user should already have their information
+        # populated from the user context provider, no need to add it
+        message_info['receiver'] = receiver.to_dict(excludes=['profile', 'password_hash'])
+        message_info['receiver']['profile'] = receiver.profile.to_dict(excludes=['user_id', 'recipes', 'user'])
+    except NoResultFound:
+        response_dict['status'] = 404
+        response_dict['message'] = "User with id %s does not exist" % (receiver_id)
+        return jsonify(response_dict), 404
+    for field in Message.get_fields():
+        if field == "sender":
+            message_info['sender'] = user
+            continue
+        if request_dict.get(field, None) != None:
+            message_info[field] = request_dict[field]
+    # Get or create a conversation between the users
+    if Conversation.conversation_exists(user, receiver):
+        message_info['conversation'] = Conversation.get_conversation(user, receiver)
+    else:
+        message_info['conversation'] = Conversation.create(**{'user_one': user, "user_two": receiver})
+    new_message = Message.create(**message_info)
+    message_info['conversation'].add_to_relationship('messages', new_message)
+    response_dict['status'] = 201
+    response_dict['message'] = "Successfully sent message to %s" % (message_info['receiver'].profile.name)
+    return jsonify(response_dict), 201
+    
+
+@messaging_views.route("/get_received_messages", methods=["GET"])
+@authenticate
+def received_messages(**kwargs):
+    response_dict = {'messages': []}
+    user = kwargs['user']
+    # Try to get all received messages
+    # if no instances are found, catch 
+    # the exception and tell the user 
+    # there are no messages
+    try:
+        messages = Message.get_instance(multiple=True, **{'receiver': user})
+    except NoResultFound:    
+        response_dict['status'] = 204
+        response_dict['message'] = "You haven't received any messages yet"
+        return jsonify(response_dict), 204
+
+    for message in messages:
+        message_dict = message.to_dict(exclude=['receiver', 'sender', 'created_at'])
+        message_dict['sender'] = message.sender.profile.name
+        message_dict['sender_image_url'] = message.sender.profile.profile_image
+        for field, value in message.get_formatted_info.items():
+            message_dict[field] = value
+        response_dict['messages'].append(message_dict)
+
+    return jsonify(response_dict), 200
+
+@messaging_views.route('/get_sent_messages', methods=["GET"])
+@authenticate
+def sent_messages(**kwargs):
```

Comment:
Not sure why we would need this route

Reply:
I wasn't sure how we were to structure the messaging portion as a whole. I'll remove it.